File: DeJia/app/src/main/java/com/dejia/anju/api/VerificationCodeLoginApi.java
package com.dejia.anju.api;

import android.content.Context;

import com.dejia.anju.api.base.BaseCallBackApi;
import com.dejia.anju.api.base.BaseCallBackListener;
import com.dejia.anju.net.NetWork;

import java.util.Map;

//验证码登录
public class VerificationCodeLoginApi implements BaseCallBackApi {

    @Override
    public void getCallBack(Context context, Map<String, Object> maps, final BaseCallBackListener listener) {
        NetWork.getInstance().call("user", "verificationCodeLogin", maps, mData -> listener.onSuccess(mData));
    }
}

File: DeJia/app/src/main/java/com/dejia/anju/api/VerificationCodeLoginApi.kt
package com.dejia.anju.api

import android.content.Context
import com.dejia.anju.api.base.BaseCallBackApi
import com.dejia.anju.api.base.BaseCallBackListener
import com.dejia.anju.net.NetWork.Companion.instance
import com.dejia.anju.net.ServerCallback
import com.dejia.anju.net.ServerData

//验证码登录
class VerificationCodeLoginApi : BaseCallBackApi {
    override fun getCallBack(context: Context, maps: MutableMap<String, Any>, listener: BaseCallBackListener<*>) {
        instance!!.call("user", "verificationCodeLogin", maps, context, ServerCallback { mData: ServerData? -> listener.onSuccess(mData) })
    }
}